infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/RestApiRequest.java

```diff
@@ -13,150 +13,32 @@
-    final ResponseMetadata responseMetadata =
-        metadata.createResponseMetadata(
-            statusCode, Optional.ofNullable(context.header(HEADER_ACCEPT)), response);
-    context.contentType(responseMetadata.getContentType());
-    responseMetadata.getAdditionalHeaders().forEach(context::header);
-    return metadata.serialize(statusCode, responseMetadata.getContentType(), response);
-  }
+  void respondOk(Object response, CacheLength cacheLength) throws JsonProcessingException;
 
-  /** This is only used when intending to return status code without a response body */
-  public void respondWithCode(final int statusCode) {
-    context.status(statusCode);
-  }
+  void respondError(int statusCode, String message) throws JsonProcessingException;
 
-  public <T> T getPathParameter(final ParameterMetadata<T> parameterMetadata) {
-    return parameterMetadata
-        .getType()
-        .deserializeFromString(pathParamMap.get(parameterMetadata.getName()));
-  }
+  void respondWithCode(int statusCode);
 
-  public <T> Optional<T> getOptionalQueryParameter(final ParameterMetadata<T> parameterMetadata) {
-    if (!queryParamMap.containsKey(parameterMetadata.getName())) {
-      return Optional.empty();
-    }
-    return Optional.of(
-        parameterMetadata
-            .getType()
-            .deserializeFromString(
-                SingleQueryParameterUtils.validateQueryParameter(
-                    queryParamMap, parameterMetadata.getName())));
-  }
+  <T> T getPathParameter(ParameterMetadata<T> parameterMetadata);
 
-  public <T> T getQueryParameter(final ParameterMetadata<T> parameterMetadata) {
-    return parameterMetadata
-        .getType()
-        .deserializeFromString(
-            SingleQueryParameterUtils.validateQueryParameter(
-                queryParamMap, parameterMetadata.getName()));
-  }
+  <T> Optional<T> getOptionalQueryParameter(ParameterMetadata<T> parameterMetadata);
 
-  public <T> void handleOptionalResult(
-      SafeFuture<Optional<T>> future, ResultProcessor<T> resultProcessor, final int missingStatus) {
-    context.future(
-        future.thenApplyChecked(
-            result -> {
-              if (result.isPresent()) {
-                return resultProcessor.process(context, result.get()).orElse(null);
-              } else {
-                context.status(missingStatus);
-                return JsonUtil.serialize(
-                    new BadRequest(missingStatus, "Not found"), BAD_REQUEST_TYPE);
-              }
-            }));
-  }
+  <T> T getQueryParameter(ParameterMetadata<T> parameterMetadata);
 
-  @FunctionalInterface
-  public interface ResultProcessor<T> {
-    // Process result, returning an optional serialized response
-    Optional<String> process(Context context, T result) throws Exception;
-  }
+  <T> void handleOptionalResult(
+      SafeFuture<Optional<T>> future,
+      RestApiRequestImpl.ResultProcessor<T> resultProcessor,
+      int missingStatus);
```

Comment:
Can we delete this instead of adding to the interface?

Reply:
oh yes! we can